pytorch_blade/tests/disc/ops/test_activation.py

```diff
@@ -11,65 +11,49 @@
 
 import torch
 import unittest
+from tests.disc.testing_base import skipIfNoDISC
 
-from tests.disc.testing_base import DiscTestCase
+from tests.disc.testing_base import DiscTestCase, skipIfEnableTorchMlir
 from torch_blade import Config
 
 
 class TestDiscActivation(DiscTestCase):
-    def _test_activation(self, activation_func):
+    def _test_activation(self, nn_func, native_func):
         x = torch.randn([2, 4, 16, 16], device=self.device)
-        test_data = (x,)
-        self._test_cvt_to_disc(activation_func, test_data)
+        test_data = (x, )
 
-    def test_relu(self):
-        relu = torch.nn.ReLU()
-        self._test_activation(relu)
+        self._test_cvt_to_disc(nn_func, test_data)
 
         @torch.jit.script
-        def relu_func(x):
-            return torch.nn.functional.relu(x)
+        def jit_func(x):
+            return native_func(x)
+        self._test_cvt_to_disc(jit_func, test_data)
 
-        self._test_activation(relu_func)
+    def test_relu(self):
+        self._test_activation(torch.nn.ReLU(), torch.nn.functional.relu)
 
     def test_leaky_relu(self):
-        leaky_relu = torch.nn.LeakyReLU()
-        self._test_activation(leaky_relu)
-
-        @torch.jit.script
-        def leaky_relu_func(x):
-            return torch.nn.functional.leaky_relu(x)
-
-        self._test_activation(leaky_relu_func)
+        self._test_activation(torch.nn.LeakyReLU(), torch.nn.functional.leaky_relu)
 
     def test_silu(self):
-        relu = torch.nn.SiLU()
-        self._test_activation(relu)
-
-        @torch.jit.script
-        def silu_func(x):
-            return torch.nn.functional.silu(x)
-
-        self._test_activation(silu_func)
-
+        self._test_activation(torch.nn.SiLU(), torch.nn.functional.silu)
 
     def test_sigmoid(self):
-        sigmoid_func = torch.nn.Sigmoid()
-        self._test_activation(sigmoid_func)
+        self._test_activation(torch.nn.Sigmoid(), torch.nn.functional.sigmoid)
 
+    @skipIfEnableTorchMlir()
     def test_glu(self):
-        @torch.jit.script
-        def glu_func(x):
-            return torch.nn.functional.glu(x)
-
-        self._test_activation(glu_func)
+        self._test_activation(torch.nn.GLU(), torch.nn.functional.glu)
 
+    @skipIfEnableTorchMlir()
     def test_gelu(self):
-        @torch.jit.script
-        def gelu_func(x):
-            return torch.nn.functional.gelu(x)
-        self._test_activation(gelu_func)
+        self._test_activation(torch.nn.GELU(), torch.nn.functional.gelu)
+
+    @skipIfEnableTorchMlir()
+    def test_hardtanh(self):
+        self._test_activation(torch.nn.Hardtanh(), torch.nn.functional.hardtanh)
 
+    @skipIfEnableTorchMlir()
```

Comment:
What's purpose of this decorator?

Reply:
No all ut functions are supported in torch-mlir converter, I just use this decorator to pass these convertor ut that are not implemented.